### DISCUSSION TITLE
Constrained Baseline for UDP Streams

### DISCUSSION THREAD

**Discussion by cappy99** [bluenviron/mediamtx#3881] [Questions]
### Question

When publishing with ffmpeg and some hardware encoders using UDP, I am only able to view the WebRTC output stream when the encoding profile is constrained baseline (preset = ultrafast). This is not case with other protocols.

Does this occur because the UDP stream is passed directly through to WebRTC while other protocols have to be converted to the appropriate format for WebRTC?

**Comment by aler9** [comment]
Hello, data coming from any protocol is processed in the same way, except for RTSP-to-RTSP, that is the only case in which there's a passthrough.

Regarding WebRTC, the general requirement is that data should not have B-frames. It's also important that the I-frame interval is set to something reasonable (< 5 seconds).

If you still have problems, provide a way to replicate the issue, in particular a sample command to produce a (non-working) UDP stream and a (working) stream with another protocol, plus server logs.
